projects/transformers/run_utils.py

```diff
@@ -459,6 +484,286 @@ def preprocess_function(examples):
     return tokenized_datasets
 
 
+def squad_prepare_features_factory(  # noqa: C901
+                                   split,
+                                   data_args,
+                                   tokenizer,
+                                   tokenizer_kwargs,
+                                   question_column_name=None,
+                                   context_column_name=None,
+                                   answer_column_name=None,
+                                   pad_on_right=None,
+                                   ):
+    """
+    splits can be train / val / and beam search ca be yes or no
+    """
+
+    key1 = question_column_name if pad_on_right else context_column_name
+    key2 = context_column_name if pad_on_right else question_column_name
+
+    def pre_pre_process(examples):
+
+        tokenized_examples = tokenizer(
+            examples[key1], examples[key2], **tokenizer_kwargs)
+
+        return tokenized_examples
+
+    if split == "train":
+
+        def preprocess_function(examples):
+
+            examples[question_column_name] = [q.lstrip() for q in examples[question_column_name]]  # noqa: E501
+            tokenized_examples = pre_pre_process(examples)
+            sample_mapping = tokenized_examples.pop("overflow_to_sample_mapping")
+            offset_mapping = tokenized_examples.pop("offset_mapping")
+
+            # Let's label those examples!
```

Comment:
If there is a big chunk of code here we got from HF and don't really understand neither assessed it, we should instead add a comment stating that. That would be useful for a future developer to know not to change this piece of code without further understanding it.